whois-nrtm/src/main/java/net/ripe/db/whois/nrtm/NrtmConnectionPerIpLimitHandler.java

```diff
@@ -70,4 +72,23 @@ private boolean connectionsExceeded(final InetAddress remoteAddresss) {
         final Integer count = connectionCounter.increment(remoteAddresss);
         return (count != null && count > maxConnectionsPerIp);
     }
+
+    private boolean canOpenConnection(final Channel channel, final InetAddress remoteAddress) {
```

Comment:
This method is misleading, the name suggests there are no side-effects, but then it can close the connection. The method does multiple things, and it's only called once - I think it's clearer to inline it into the channelOpen method()?